src/components/NewClusterModal.js

```diff
@@ -152,6 +153,106 @@ export const NewClusterModal = withModalDrawer({ width: 675 })(class NewClusterM
+  //this function returns true for cases 2 & 3 in this diagram
+  canUpdate() {
+    const { currentCluster } = this.props
+
+    if (!currentCluster) return false
+
+    const currentClusterConfig = currentCluster.machineConfig
+    const userSelectedConfig = this.getMachineConfig()
+
+    const cantWorkersUpdate = currentClusterConfig.numberOfWorkers !== userSelectedConfig.numberOfWorkers &&
+      (currentClusterConfig.numberOfWorkers < 2 || userSelectedConfig.numberOfWorkers < 2)
+
+    const hasUnUpdateableResourceChanged =
+      currentClusterConfig.workerDiskSize !== userSelectedConfig.workerDiskSize ||
+      currentClusterConfig.workerMachineType !== userSelectedConfig.workerMachineType ||
+      currentClusterConfig.numberOfWorkerLocalSSDs !== userSelectedConfig.numberOfWorkerLocalSSDs
+
+    const hasWorkers = currentClusterConfig.numberOfWorkers >= 2 || currentClusterConfig.numberOfPreemptibleWorkers >= 2
+    const hasWorkersResourceChanged = hasWorkers && hasUnUpdateableResourceChanged
+
+    const hasDiskSizeDecreased = currentClusterConfig.masterDiskSize > userSelectedConfig.masterDiskSize
+
+    const cantUpdate = cantWorkersUpdate || hasWorkersResourceChanged || hasDiskSizeDecreased || this.hasImageChanged() || this.hasStartUpScriptChanged()
+    return !cantUpdate
+  }
+
+  hasChanges() {
+    const { currentCluster } = this.props
+    if (!currentCluster) return true
+
+    //TODO: this _.pickBy will need to change if the UI-side machineConfig starts tracking the cloud service (which it will once Leo exposes the ability to create GCE runtimes)
+    const currentClusterWithoutService = _.pickBy((v, k) => k !== 'cloudService', currentCluster.machineConfig)
+    const hasMachineConfigChanges = !_.isMatch(currentClusterWithoutService, this.getMachineConfig())
+
+    return hasMachineConfigChanges || this.hasImageChanged() || this.hasStartUpScriptChanged()
+  }
+
+  //returns true for case 3 in this diagram: https://drive.google.com/file/d/1mtFFecpQTkGYWSgPlaHksYaIudWHa0dY/view
+  isStopRequired() {
+    const { currentCluster } = this.props
+
+    const currentClusterConfig = currentCluster.machineConfig
+    const userSelectedConfig = this.getMachineConfig()
+
+    const isMasterMachineTypeChanged = currentClusterConfig.masterMachineType !== userSelectedConfig.masterMachineType
+
+    const isClusterRunning = currentCluster.status === 'Running'
+
+    return this.canUpdate() && isMasterMachineTypeChanged && isClusterRunning
+  }
+
+  getRunningUpdateText() {
+    return this.isStopRequired() ?
+      p([
+        'Changing the machine type (increasing or decreasing the # of CPUs or Mem) results in an update that requires a ',
+        b(['restart']),
+        ' of your runtime. This may take a 3-5 minutes. Would you like to proceed? ',
+        b(['(You will not lose any files.)'])
+      ]) :
+      p([
```

Comment:
This should also get 1.5 line height

Reply:
Hmm... I'm going to leave it because there are other p() tags that don't have it, and I'm unsure if the desire is for all of them to have it, or just the warning ones, which I feel is another discussion and would require a bit more refactoring.

Reply:
It's not so much a p() thing as it is that when there is a block of text the line height should be 1.5 as 1.5 line height is the general accepted accessible line height and with the accessibility audit impending it's something we are trying to follow